### ISSUE TITLE
Sound when line goes over editor.rulers setting

### ISSUE THREAD

**Issue by justas2481** [microsoft/vscode#150646]
Issue Type: <b>Feature Request</b>

Since now there are sounds to indicate an error (for people who use screen readers), it would be really helpful to have an accessible way of knowing that lines within source code are too long.
It could give some different audio tone than it does for an error, just to quickly identify that. Furthermore, some shortcut to search for a lines that are longer than ruler / line length value would be benefitial.
It's really important for the blind, because we usually do not care about too long lines (unless they're too long for ourselves). When sighted people are involved into the same code, we should keep it consistent and easily readable and this feature would help to achieve it.

VS Code version: Code 1.67.2 (c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5, 2022-05-17T18:15:52.058Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No
Remote OS version: Linux x64 5.15.0-33-generic

**Comment by snufas** [comment]
Yes I agree @justas2481  Some kind of alert would be  nice indeed,  particularly if the formatter is not fixing those on its own, And it is not easy to apply all the formatting requirements if they are not properly indicated for screen reader users.

@isidorn  Any thoughts regarding this feature implementation? It is definitely useful and probably not too tricky to implement. I suppose there is a good reason why VS code has a vertical ruler.

**Comment by isidorn** [comment]
Great idea and thanks for opening this issue.
I believe this should work as following: if the user has set "editor.rulers" setting, and if a line is over the value of the setting we should produce a sound.

@hediet what do you think?

To me it sounds like a nice use case and I could talk to the new sound designer for this.

**Comment by justas2481** [comment]
@isidorn That would be a good implementation. Maybe it might be considerable to announce how many chars cursor / carriage is over the rular limit when navigating char by char, but this might not be necessary (needs deeper thinking on it).
From my point of view, different ruler per language / file extension could be useful as well, though I'm not so sure how it works for the sighted in this case.

**Comment by justas2481** [comment]
@isidorn @hediet I guess some shortcut key to let screen reader to announce current char in the line (it's position) to avoid agressivness would be a solution. So in general when user goes through source code by using up/down arrows and finds a line in which there are more chars then the ruler setting, it indicates it by some sort of a short beep or something. By then, user already knows about the problem, so hits a shortcut to check the length which might say something like "char 14 / 93" which means that this line contain 93 characters and the current one which caret is standing on is 14th character in that line.

**Comment by justas2481** [comment]
@isidorn is this considered to be implemented?

**Comment by isidorn** [comment]
@justas2481 not yet. We are still in the collecting feedback phase and considering this feature.

**Comment by mikolysz** [comment]
As a screen reader user, I fully support something like this. I think that, instead of having a list of offending lines, we should have two commands, one to go to the next offending line, and another to go to the previous one.

A command which announces the current column, as well as the number of total characters in the line, would also be useful, not just for this problem, but also for properly aligning text with white space.

**Comment by isidorn** [comment]
@mikolysz that is a fair request, but is something that could be done by an extension. We expose all of this API, and I think it should not be too much work. The extension should read the editor setting to know where the limit is, and the introduce the commands that would just navigate across the editor. When you find time maybe you could give this a try? https://code.visualstudio.com/api/get-started/your-first-extension

Because we try to have a very minimalistic product, whenever things can be done by extensions we highly recommend that it is done that way.

**Comment by hediet** [comment]
> We should just figure out if that would be a bit too aggressive for editor.rulers users.

I think this would be the problem for all warnings, so we should rather work on making warnings less aggressive.

Lets close this as extension candidate for now and revisit this issue when it turns out that warnings are too aggressive.

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by derekriemer** [comment]
Reopening because this can't currently be implemented with an extension.
See https://github.com/microsoft/vscode/issues/175986, we can't contribute signal sounds.
